R/util.R

```diff
@@ -5,6 +5,18 @@
     x
 }
 
+findNonces <- function(s) {
+  # First group is the id passed to shiny::registerDataObj(), second group is
+  # the unique 'nonce' identifier generated by Shiny.
+  nonceRe <- '/dataobj/([^\\?]+)\\?w=[^&]*&nonce=([0-9a-f]+)'
+  df <- as.data.frame(stringr::str_match_all(s, nonceRe), stringsAsFactors = FALSE)
+  # We order by data object id here to ensure nonce numbers are stable even if
+  # the URLs we found appear in a different order in the message during
+  # playback. This is defensive: JSON object key order is unlikely to be
+  # different for the same version of jsonlite.
+  df <- df[order(df$X2),]
+  df[,'X3']
```

Comment:
add a third argument of `drop = TRUE` or restructure data.frame call to be `df$X3` or `df[["X3"]]`